modules/artifact/src/main/java/org/mule/runtime/module/artifact/classloader/FilteringArtifactClassLoader.java

```diff
@@ -65,12 +77,27 @@ public FilteringArtifactClassLoader(ArtifactClassLoader artifactClassLoader, Cla
 
   @Override
   public URL getResource(String name) {
-    if (filter.exportsResource(name)) {
-      return getResourceFromDelegate(artifactClassLoader, name);
+    if (isServiceResource(name)) {
+      String serviceInterface = getServiceInterfaceFromResource(name);
+
+      Optional<ExportedService> exportedService =
```

Comment:
add a log of what entries were found and which will be used, for torubleshooting.